actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import logging
+from pathlib import Path
+from typing import List, NamedTuple, Optional, Union
+
+import aicsimageio
+import aicsimageprocessing as proc
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import matplotlib.image as mpimg
+import matplotlib.pyplot as plt
+from aics_dask_utils import DistributedHandler
+from aicsimageio import AICSImage, transforms
+from aicsimageio.writers import OmeTiffWriter
+from datastep import Step, log_run_params
+from imageio import imwrite
+
+from ...constants import Channels, DatasetFields
+from ...utils import dataset_utils, image_utils
+from ..single_cell_features import SingleCellFeatures
+from ..single_cell_images import SingleCellImages
+
+plt.style.use('dark_background')
+###############################################################################
+
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+REQUIRED_DATASET_FIELDS = [
+    DatasetFields.CellId
+]
+
+class DiagnosticSheetResult(NamedTuple):
+    cell_id: Union[int, str]
+    diagnostic_sheet_save_path: Optional[Path] = None
+
+
+class DiagnosticSheetError(NamedTuple):
+    cell_id: Union[int, str]
+    error: str
+
+
+###############################################################################
+
+
+class MakeDiagnosticSheet(Step):
+    def __init__(
+        self,
+        direct_upstream_tasks: List["Step"] = [SingleCellImages]
+    ):
+        super().__init__(direct_upstream_tasks=direct_upstream_tasks)
+
+
+    @staticmethod
+    def _make_group_plot(
+        dataset: pd.DataFrame,
+        group_by: str,
```

Comment:
these seem to appear together, why do you need to set both a key and value?  can you give an example of what both might be?